openedx/features/edly/api/v1/views/enrollment_count.py

```diff
@@ -0,0 +1,54 @@
+from rest_framework import status, viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from common.djangoapps.student.models import CourseEnrollment
+from openedx.core.djangoapps.catalog.utils import get_programs
+
+
+class EdlyProgramEnrollmentCountViewSet(viewsets.ViewSet):
+    """
+    **Use Case**
+
+        Get the number of enrollments for a program.
+
+    **Example Request**
+
+        GET /api/v1/programs/enrollment_count/{program_uuid}
+
+    **GET Parameters**
+
+        * program_uuid: The UUID of the program.
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+
+        The HTTP 200 response has the following values.
+
+        * enrolled_users_count: The number of enrollments for the program. 
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def list(self, request, *args, **kwargs):
+        program_uuid = kwargs.get('program_uuid')
+        program = get_programs(uuid=program_uuid)
+
+        if not program:
+            return Response({'detail': 'Program not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        courses = program.get('courses', [])
+        enrolled_user_ids = set()
+
+        for course in courses:
+            course_runs = course.get('course_runs', [])
+            for course_run in course_runs:
+                enrollments = CourseEnrollment.objects.filter(
+                    course_id=course_run['key']).values_list('user', flat=True)
+                enrolled_user_ids.update(enrollments)
+
+        response_data = {
+            'enrolled_users_count': len(enrolled_user_ids),
+        }
+
+        return Response(response_data, status=status.HTTP_200_OK)
```

Comment:
You can use the retrieve method for a single program.

Won't enrollments = CourseEnrollment.objects.filter(course_id__in=course_run_ids).values_list('user', flat=True).distinct().count() do the job here?

Reply:
updated